### ISSUE TITLE
memory leaks

### ISSUE THREAD

**Issue by hktalent** [projectdiscovery/nuclei#3301]
```console
cat $HOME/MyWork/scan4all/json/paypal.com.json|sed 's/=.*//g'|sort -u|nuclei -sa -t $PWD/config/nuclei-templates -no-strict-syntax -severity critical,high,medium -type http,network,websocket,dns,ssl -report-config $PWD/config/nuclei_esConfig.yaml  -ztls -config-directory ${PWD}/config/nuclei -interactions-cache-size 5000 -interactions-eviction 60 -interactions-poll-duration 5 -interactions-cooldown-period 5 -max-host-error 5 -duc -json -o paypal_nuclei.json
```

<img width="748" alt="image" src="https://user-images.githubusercontent.com/18223385/218058179-433307d2-c567-4411-8cb3-d644b7088c1b.png">

**Comment by tarunKoyalwar** [comment]
@hktalent -> https://nuclei.projectdiscovery.io/nuclei/get-started/#mass-scanning-using-nuclei

**Comment by hktalent** [comment]
<img width="683" alt="image" src="https://user-images.githubusercontent.com/18223385/218234309-46cb3139-e745-4094-b23a-e9b770e073aa.png">

**Comment by hktalent** [comment]
<img width="604" alt="image" src="https://user-images.githubusercontent.com/18223385/218280735-ee128baf-c731-475d-be97-026406899492.png">

**Comment by tarunKoyalwar** [comment]
@hktalent , can you share the exact command you are running , its difficult to guess what the cause is with screenshots

**Comment by Mzack9999** [comment]
@tarunKoyalwar I think the command should be the following one (with input size equal to the PayPal subdomain list and, I guess, a standard set of nuclei templates):

```go
nuclei -sa -t $PWD/config/nuclei-templates -no-strict-syntax -severity critical,high,medium -type http,network,websocket,dns,ssl -report-config $PWD/config/nuclei_esConfig.yaml  -ztls -config-directory ${PWD}/config/nuclei -interactions-cache-size 5000 -interactions-eviction 60 -interactions-poll-duration 5 -interactions-cooldown-period 5 -max-host-error 5 -duc -json -o paypal_nuclei.json
```

**Comment by hktalent** [comment]
@Mzack9999 
nuclei templates
target:11993

**Comment by hktalent** [comment]
@Mzack9999 
> I'm having some problems reproducing it with the latest release. Are you still facing this issue? Is it reproducible in a consistent manner? Running the tool multiple times with the same options yields to memory outage?

You start a process and then loop through starting several nuclei instances. Of course, you can limit the number of concurrent instances, for example, 10. If you run it continuously overnight, you will find out where the memory leak is.

**Comment by luzhaocong** [comment]
I encountered the same issue. I randomly scanned a list of Class B IP addresses using the following command: nuclei -stream -scan-strategy host-spray -rl 500 -bs 200 -c 10 -timeout 2 -stats -metrics-port 9092 -no-httpx -u xxx.xxx.0.0/16 -o result.txt. I ran this command continuously for a day on a Linux server with 4 cores and 8 GB of RAM. The memory usage kept increasing until the server eventually crashed.
